src/number_theory/number_field.lean

```diff
@@ -249,4 +251,113 @@ end
 
 end roots
 
+section bounded
+
+open finite_dimensional polynomial set
+
+variables (K : Type*) [field K] [number_field K]
+variables (A : Type*) [normed_field A] [is_alg_closed A] [normed_algebra ℚ A]
+
+lemma finite_of_norm_le (B : ℝ) :
+  {x : K | is_integral ℤ x ∧ ∀ φ : K →+* A, ∥ φ x ∥  ≤ B}.finite :=
+begin
+  classical,
+  let S := finset.bUnion
+    (finset.product (finset.range (finrank ℚ K + 1)) (finset.range (finrank ℚ K + 1)))
+    (λ x, ( { B ^ (x.1 - x.2) * (x.1.choose x.2) } : finset ℝ)),
+  let C := nat.ceil (S.max' _),
+  swap,
+  { exact finset.bUnion_nonempty.mpr
+    ⟨⟨0 , 0⟩, finset.mem_product.mpr ⟨finset.mem_range_succ_iff.mpr (zero_le _),
+      finset.mem_range_succ_iff.mpr (zero_le _)⟩, finset.singleton_nonempty _⟩, },
+  suffices :
+    (⋃ (f : polynomial ℤ)
+       (hf : f.nat_degree ≤ finrank ℚ K ∧ ∀ i, |f.coeff i| ≤ C),
+       ((f.map (algebra_map ℤ K)).roots.to_finset : set K)).finite,
+  { refine this.subset _,
+    clear this,
+    intros x hx,
+    have h_map_rat_minpoly := minpoly.gcd_domain_eq_field_fractions' ℚ hx.1,
+    have h_same_deg_minpoly : (minpoly ℚ x).nat_degree = (minpoly ℤ x).nat_degree,
+    { rw h_map_rat_minpoly, convert nat_degree_map_eq_of_injective _ _,
+      exact (algebra_map ℤ ℚ).injective_int, },
+    have h_bdd_degree : (minpoly ℚ x).nat_degree ≤ finrank ℚ K,
+    { refine le_of_eq_of_le
+        (intermediate_field.adjoin.finrank (is_integral_of_is_scalar_tower _ hx.1)).symm _,
+      exact ℚ⟮x⟯.to_subalgebra.to_submodule.finrank_le, },
+    have h_roots_bdd_minpoly : ∀ z ∈ (map (algebra_map ℚ A) (minpoly ℚ x)).roots, ∥ z ∥  ≤ B,
+    { intros z hz,
+      suffices : ∃ (φ : K →+* A), φ x = z,
+      { obtain ⟨φ, rfl⟩ := this, exact (hx.2 φ), },
```

Comment:
If I remember correctly @ericrbg made a modification to avoid having to write this, but I am not sure.

Reply:
See https://leanprover-community.github.io/mathlib_docs/tactics.html#rsuffices